src/Hint/NumLiteral.hs

```diff
@@ -34,10 +36,16 @@ import Idea (Idea, suggest)
 
 numLiteralHint :: DeclHint
 numLiteralHint _ modu =
-  if NumericUnderscores `elem` extensions (modComments modu) then
+  -- TODO: there's a subtle bug when the module disables `NumericUnderscores`.
+  -- This seems pathological, though, because who would enable it for their
+  -- project but disable it in specific files?
+  if NumericUnderscores `elem` activeExtensions then
      concatMap suggestUnderscore . universeBi
   else
      const []
+  where
+    moduleExtensions = toList (extensions $ modComments modu)
+    activeExtensions = hlintExtensions modu <> toList moduleExtensions
```

Comment:
I think the `toList` here is redundant - you've already called `toList` inside `moduleExtensions`, right?